### PR TITLE
Add support for mimic joints

### PR DESCRIPTION
**Pull request by costashatz**
This PR adds support for mimic joints (see #733). It does so by creating a new actuator type (MIMIC) and inserting one constraint for each mimic joint. In short:

- Mimic joints are treated as velocity-controlled dynamic joints; similar to servo motors
- A mimic joint will try to follow the (multiplied and offset-ed) position of a "master" joint
- The `JointProperties` struct now holds three additional variables: `mMimicJoint` (pointer to "master" joint), `mMultiplier` and `mOffset` (these are only used in mimic joints and ignored for all other types of joints)
- DartLoader (i.e., URDF loader) now correctly loads URDF files with mimic joints. Is there a similar functionality/tag in `skel` files (I know that in `sdf` there's no such thing)?
- Cloning a skeleton, adjusts the mimic joint pointers

This is working for me nicely so far..

Let me know what you think and then I can proceed in cleaning the code (add a few more warnings and check it a bit more). ~Do you think it will be good to add a unit-test?~ I also added 2 unit tests (one for DartLoader and one for the mimic constraint)..

### CODE REVIEW

File: dart/dynamics/detail/JointAspect.hpp [hunk 1]
```diff
@@ -117,14 +125,23 @@ struct JointProperties
...
              const Eigen::Isometry3d& _T_ParentBodyToJoint =
                                  Eigen::Isometry3d::Identity(),
              const Eigen::Isometry3d& _T_ChildBodyToJoint =
                                  Eigen::Isometry3d::Identity(),
              bool _isPositionLimitEnforced = false,
-             ActuatorType _actuatorType = DefaultActuatorType);
+             ActuatorType _actuatorType = DefaultActuatorType,
+             Joint* _mimicJoint = nullptr,
+             double _multiplier = 1.0,
+             double _offset = 0.0);
```
Comment (mxgrey) [inline dart/dynamics/detail/JointAspect.hpp]:
```suggestion
             double _mimicMultiplier = 1.0,
             double _mimicOffset = 0.0);
```

to make it clear that these are properties specifically related to the mimic joint actuator type.

**Comment by costashatz** [comment]
@jslee02 your changes seem great! Integrated them and I updated the changelog..

**Comment by costashatz** [comment]
Is there any reason that you are not merging this one? Thanks!

**Comment by jslee02** [comment]
I was waiting for a second review. Let me merge this now.

**Pull request merged**